src/UnoFlipModel.java

```diff
@@ -12,368 +12,228 @@
+            this.deck.putCard(this.topCard);
+            this.topCard = this.deck.takeCard();  //redraw the topCard
         }
 
-        if(topCard.getRank().ordinal()>8) {
-            //only execute sequence if action card, if number don't do anything
-            turnSeqs.get(topCard.getRank().ordinal()).executeSequence(topCard);
-        }
-        else{
-            currentColour = topCard.getColour();
-            currentRank = topCard.getRank();
-            status = " ";
-            notifyViews(); // Notifying view here since it is the last step in the initialization of the game,
+        //if first card drawn form deck is an action card (non-number card)
+        if(this.topCard.getRank().ordinal() > Card.RANK_NUMBER_CARDS) {
+            this.turnSeqs.get(this.topCard.getRank().ordinal()).executeSequence(this.topCard); //execute sequence if action card
+
+        //if number card drawn
+        } else{
+            this.currentColour = this.topCard.getColour();
+            this.currentRank = this.topCard.getRank();
+            this.status = STATUS_STANDARD;
+            notifyViews();
         }
     }
 
 
     /**
-     * Method notifyViews is meant to notify subscribers view about any changes that will effect the view of the UnoFlip Game
-     *
+     * Method notifyViews is meant to notify subscribers view about any changes that will affect the view of the UnoFlip Game
      */
     public void notifyViews(){
-        if(!views.isEmpty()) {
-            //send UnoFlipEvent to view.
 
-            if (topCard.isWild() && !status.equals("INNOCENT: NEXT PLAYER DRAWS 4 CARDS") && !status.equals("GUILTY:YOU DRAW 2 CARDS") && !dontAsk){
-                status = currentColour.toString();
-                for( UnoFlipView view: views ) {
-                    view.handleUnoFlipStatusUpdate( new UnoFlipEvent(this, getCurrentPlayer().getName(), topCard.toString(), getCurrentPlayer().toString(),status,(this.currentRank == Card.Rank.WILD || this.currentRank == Card.Rank.WILD_DRAW_2)));
-                }
-            } else{
-                for( UnoFlipView view: views ) {
-                    view.handleUnoFlipStatusUpdate( new UnoFlipEvent(this, getCurrentPlayer().getName(), topCard.toString(), getCurrentPlayer().toString(),status ,this.currentRank == Card.Rank.WILD ));
+        //make sure there are views in the view arraylist to send UnoFlipEvents to.
+        if(!this.views.isEmpty()){
+
+            //sending events to the view to update depending on different situations that occur within the game
+            //if wild draw 2 card and next player declines to challenge
+            if(this.topCard.isWild() && !this.status.equals(STATUS_CHALLENGE_INNOCENT) && !this.status.equals(STATUS_CHALLENGE_GUILTY) && !this.dontAsk){
+
+                this.status = this.currentColour.toString(); //set status as the current colour chosen by the player (ex: RED)
+
+                for( UnoFlipView view: this.views ) {
```

Comment:
I think we should not do this in this branch. since reason for the first check is for a wild Draw 2/ challenge. So i believe that since tharusha is making a new branch to fix the challenge implementation (which he will most likely make edits to this method) we should handle its implementation in that branch.

Reply:
this isn't changing anything about the implementation. make this change please